release/Magarena/scripts/Din_of_the_Fireherd.groovy

```diff
@@ -0,0 +1,40 @@
+def action = {
+    final MagicGame game, final MagicEvent event ->
+            event.processTargetPlayer(game, {
+                final MagicPlayer player = event.getPlayer();
+                final int BC = BLACK_CREATURE_YOU_CONTROL.filter(player).size();
+                game.logAppendValue(player,BC)
+                for (int i = 0; i < BC; i++) {
+                    game.addEvent(new MagicSacrificePermanentEvent(event.getSource(), it, SACRIFICE_CREATURE));
+                }
+                final int RC = RED_CREATURE_YOU_CONTROL.filter(player).size();
+                game.logAppendValue(player,RC)
+                for (int i = 0; i < RC; i++) {
+                    game.addEvent(new MagicSacrificePermanentEvent(event.getSource(), it, SACRIFICE_LAND));
+        }
+    });
+}
+
+[
+    new MagicSpellCardEvent() {
+        @Override
+        public MagicEvent getEvent(final MagicCardOnStack cardOnStack,final MagicPayedCost payedCost) {
+            return new MagicEvent(
+                cardOnStack,
+                this,
+                "PN create a 5/5 black and red Elemental creature token."
+            );
+        }
+        @Override
+        public void executeEvent(final MagicGame game, final MagicEvent event) {
+            game.doAction(new PlayTokensAction(event.getPlayer(), CardDefinitions.getToken("5/5 black and red Elemental creature token"), 1));
+            game.doAction(new PutItemOnStackAction(new MagicTriggerOnStack(new MagicEvent(
```

Comment:
When solving the targetting below, I have encountered a problem that the sacrifice effect does not "see" the token just created.... I guess that is why the stack-putting was here originally. This might be a bit trickier to solve nicely - I will remove this card from this request for now and I'll do it properly later.